### PR TITLE
Cache client not set for paginated full folder item requests

### PR DESCRIPTION
**Pull request by adempsey**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @adempsey has signed the CLA. Thanks for the pull request!

**Review by jlawton — APPROVED** [review]
👍

File: BoxContentSDK/BoxContentSDK/Requests/BOXFolderItemsRequest.m [hunk 1]
```diff
@@ -179,6 +179,7 @@ - (void)performPaginatedRequestWithCached:(BOXItemArrayCompletionBlock)cacheBloc
                                   inRange:(NSRange)range
 {
     BOXFolderPaginatedItemsRequest *paginatedRequest = [[BOXFolderPaginatedItemsRequest alloc] initWithFolderID:self.folderID inRange:range];
+    paginatedRequest.cacheClient = self.cacheClient;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFolderItemsRequest.m:182]: Is the behavior correct? Does it call the cache client with the expected sub range?

Comment (adempsey) [reply BoxContentSDK/BoxContentSDK/Requests/BOXFolderItemsRequest.m:182]: Yes. `BOXFolderPaginatedItemsRequest` uses a different callback method than this class, `cacheFolderPaginatedItemsRequest:withItems:limit:offset:error:`

**Pull request merged**